### ISSUE TITLE
🚀 Feature: `scripts-name-casing`: add an option to ignore script names

### ISSUE THREAD

**Issue by andreww2012** [JoshuaKGoldberg/eslint-plugin-package-json#2050]
### Feature Request Checklist

- [x] I have [searched for related issues](https://github.com/michaelfaith/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

As #2047 showed, packages managers might introduce special script name conventions which might not end up using the kebab-case that the rule enforces.

For example, recently pnpm introduced ["hidden scripts"](https://pnpm.io/scripts#hidden-scripts) (starting with a dot) and a special [`pnpm:devPreinstall`](https://pnpm.io/scripts#pnpmdevpreinstall) script which both fail the validation:

```json
{
  "pnpm:devPreinstall": "echo foo",
  ".bar": "echo bar"
}
```

To be able to make exceptions at any time, I propose adding a new `ignore` option, which should take an array of strings. If a string starts with `/` and ends with `/<optional regex flag(s)>`, it's treated as a regex; otherwise as a plain string (on the edge case: this would be super weird to ignore a script name starting literally with /, but that would still be possible to ignore using the regex form).

The example of configuring the rule to ignore the mentioned script names:

```ts
const rules = {
  'package-json/scripts-name-casing': [
    2,
    {
      ignore: ['pnpm:devPreinstall', '/^\\.[a-z0-9]+(-[a-z0-9]+)*$/'],
    },
  ],
};
```

**Comment by michaelfaith** [comment]
As I mentioned in the original issue, I'm in favor of this feature 👍.  The only part I'd like to bikeshed a bit, is the regex handling.  Rather than having regex patterns and explicit names being part of the same option (which then requires people using the `/ /` to signal its a regex pattern, I think it might be cleaner to have those be two separate options.  Perhaps 
- `ignoreNames`: a list of specific script names to ignore
- `ignorePatterns`: a list of regex patterns to use for ignores (without surrounding `/ /`)

How do you feel about that?

**Comment by andreww2012** [comment]
To be honest, I saw this ignore option handling used in other plugins, and I kind of liked it. So I do your suggestion. The only thing we lose by separating options is the ability to specify regex flags. If that's something you think we can live without, I don't really have any objections to your suggestion 👍

**Comment by michaelfaith** [comment]
Sounds good!  Sounds like we're aligned.  Marking as "accepting prs" 👌
